src/blake2/blamka-round-opt.h

```diff
@@ -176,5 +177,152 @@ static BLAKE2_INLINE __m128i fBlaMka(__m128i x, __m128i y) {
                                                                                \
         UNDIAGONALIZE(A0, B0, C0, D0, A1, B1, C1, D1);                         \
     } while ((void)0, 0)
+#else /* __AVX2__ */
 
-#endif
+#include <immintrin.h>
+
+#define rotr32(x)   _mm256_permutevar8x32_epi32(x, _mm256_setr_epi32(1, 0, 3, 2, 5, 4, 7, 6))
```

Comment:
You could also use `_mm256_shuffle_epi32(x, _MM_SHUFFLE(2, 3, 0, 1))` here. That instruction takes an 8-bit immediate instead of loading a 256-bit constant, so it might be faster. (I didn't measure it, though)